internal/cache/delete.go

```diff
@@ -23,16 +24,21 @@ func deleteUserFromGroup(buckets map[string]bucketWithName, uid, gid int) error
 		return nil
 	}
 
-	// Delete empty group
+	// We now need to delete this group with no remaining user.
+	// We need the group.Name to delete from groupByName bucket.
 	group, err := getFromBucket[groupDB](buckets[groupByIDBucketName], gid)
 	if err != nil {
 		return err
 	}
-
-	// RW transactions can't fail
-	_ = buckets[groupToUsersBucketName].Delete([]byte(strconv.Itoa(group.GID)))
-	_ = buckets[groupByIDBucketName].Delete([]byte(strconv.Itoa(group.GID)))
-	_ = buckets[groupByNameBucketName].Delete([]byte(group.Name))
+	if err = buckets[groupToUsersBucketName].Delete([]byte(strconv.Itoa(group.GID))); err != nil {
+		panic(fmt.Sprintf("CRITICAL: unrecoverable state: could not delete group %q from %q: %v", group.Name, groupToUsersBucketName, err))
```

Comment:
really, tag it as "programming error", so that we know we are not going to send this to the finale user.

Reply:
And you removed the comment to: which stilll explains the panic I guess.

Reply:
As in `PROGRAMMING ERROR: could not...` or as in `CRITICAL: programming error: could not...`?

Reply:
I would say do like the rest of the code